runtime/lua/vim/shared.lua

```diff
@@ -801,7 +802,7 @@ end
 ---@return string String with whitespace removed from its beginning and end
 function vim.trim(s)
   vim.validate('s', s, 'string')
-  return s:match('^%s*(.*%S)') or ''
+  return _trim_pattern:match(s)
```

Comment:
will it return empty string if there's no match? or is `or ''` needed?

Reply:
Yes. `{(%s* %S+)*}`.
`{}` - means return capture, not index
`()*` returns 0 or as many as possible matches. And if there is 0 it will return empty string.